documentation/eng_sys.md

```diff
@@ -0,0 +1,50 @@
+# Engineering System Checks
+
+* [Build and Test](#build-and-test)
+* [Analyze Stages](#analyze-stages)
+
+## Build and Test
+
+Our build system runs PR changes against Go versions 1.14 and 1.15 on both Windows and Linux
+
+## Analyze
+
+### Link Verification Check
+Verifies all of the links are valid in your README files. This step also checks that locale codes in links are removed. If this is failing first check if you have locale codes (ie. `en-us`) in your links, then check to see if the link works locally.
+
+If you are trying to add a link that will exist in the next PR (ie. you are adding a samples README or migration guide), you can use an `aka.ms` link or use a temporary link (ie: `https://microsoft.com`) and create a follow-up PR to correct the temporary link.
+
+### Lint
+Linting will not fail a CI check, but it will show possible issues with your design. Some of the most common are:
+* `errcheck`: An error was returned but it was not checked to be `nil`
+* `varcheck`: A variable is unused
+* `deadcode`: A struct or method is unused
+* `ineffasign`: An ineffectual assignment, the variable is not used after declaration.
+
+For more information about the linters run checkout the [golangci website][golangci_website]
+
+To run this locally, first install the tool with:
+```bash
+go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.41.1
```

Comment:
As of Go 1.17, the preferred way to install binary tools is `go install`, see https://golang.org/doc/go1.17#go-get